src/pudl_archiver/archivers/eia191.py

```diff
@@ -0,0 +1,98 @@
+"""Download EIA 191 data."""
+import logging
+import zipfile
+from collections.abc import Iterable
+
+import pandas as pd
+
+from pudl_archiver.archivers.classes import (
+    AbstractDatasetArchiver,
+    ArchiveAwaitable,
+    EIANaturalGasData,
+    ResourceInfo,
+)
+from pudl_archiver.frictionless import ZipLayout
+from pudl_archiver.utils import add_to_archive_stable_hash
+
+logger = logging.getLogger(f"catalystcoop.{__name__}")
+
+
+class Eia191Archiver(AbstractDatasetArchiver):
```

Comment:
As for sharing code between 191 and 757A, the only real differences I see between the two archivers are: the base url/`form` that get passed to `get_datasets`, the filter for `Monthly` in 191 `get_resources`, and the sorting by `company_name` and `field_name` for 191. With these minimal differences in mind, my inclination would be to create a base class that both of these archivers inherit from. The base class would inherit from `AbstractDatasetArchiver` and implement `get_datasets` and `get_year_resource`, leaving out the sorting that done in the 191 implementation (could this be done later in the ETL?). This way the `Eia191Archiver` and `Eia757AArchiver` could just set their `name` and `base_url`, then implement `get_resources` where 191 can filter for monthly data.